### ISSUE TITLE
Sscore2 submission

### ISSUE THREAD

**Issue by harrisgm** [Bioconductor/Contributions#1190]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/harrisgm/Sscore2

Confirm the following by editing each check box to '[x]'

- [X] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [X] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [X] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [X] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [X] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [X] The 'devel' branch for new packages and features.
- [X] The stable 'release' branch, made available every six
      months, for bug fixes.
- [X] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/

**Comment by LiNk-NY** [comment]
Hi Guy, @harrisgm 

Thank you for your submission.
Unfortunately, we are unable to accept your package in its current form. 
Please see the review below and respond to the pertinent comments. 

Best regards,
Marcel

---

# Sscore2 #1190

## DESCRIPTION

* Packages should be in `Imports` and not `Depends`
* Remove dependencies that are not being used (`dplR`)

## NAMESPACE

* Remove the `exportPattern` directive and be intentional with your exports

## Comments before further review

* Please make sure you are interfacing with established Bioconductor data
structures, otherwise the package seems to fall out of scope of the project
* The code should follow the Bioconductor package guidelines:
http://bioconductor.org/developers/package-guidelines
* Given that Affy microarrays have been replaced by newer high-throughput
sequencing tech, what are the advantages of using `Sscore2` versus `Sscore`?
* Consider renaming the package (or contributing to the original package) so
that the package does not appear to be a continuation of `Sscore`.

**Comment by harrisgm** [comment]
Hi Marcel Ramos,

I'm very sorry for the delayed response, I somehow missed this email on
August 7th.

I have been working on the failed build issues and simplifying the
annotation structure of the data.

In response to the reviewer comments:
1. I can resolve the "Description" and "Namespace" issues this evening.
2. The Sscore2 package is designed to work efficiently with ClariomD/MTA
arrays to provide exon-level and gene-level differential expression using
probe-level data and GC-based background corrections.  The package is
entirely new in terms of code and operation, it just expands on the
mathematical principles established in the first package.  Analysis tools
for HTA/MTA/ClariomD and ClariomS (arrays that are still sold and used) are
lacking and this packages fills that void.  And it allows for more complex
and fine tuned dissection of the different PSR/TC IDs types included in the
HTA/MTA/ClariomD (see "locus_type" discussion below and my needs for
additional probe-level annotation information)
3. We could certainly consider renaming the package if necessary.
4.  The most important consideration is the use of our "probefiles", which
are efficient tables that are generated from Affymetrix CDF (for ClariomD)
or PGF (for ClariomS) files (publicly available on both Affymetrix's site
and Thermo-Fishers).  We need to have the GC.count of every probe included
in the probe level data. We also use the Affymetrix na36 annotation data to
get the "locus_type" information that provided for the probesetIDs or
transcriptClusterIDs, and include it in the probe-level data.  This allows
the user to select which PSRid/TCid types are included in the analysis
(since there are more noisy "noncoding" PSRids/TCids than there are
"Coding" IDs on the ClariomD arrays).  We also made annotation files with
Affymetrix na36 data.

I can easily change the annotation assignments to use the annotation.db
packages provided in bioconductor.

The probe-level data is more difficult since I do not believe there is an
annotation resource on Bioconductor that includes the "locus_type" info or
the "GC count" info for each individual probe.

Should I submit my affymetrix generated probe-level datafiles as annotation
pacakges in order to solve this issue?

Sorry again for the delayed response.  Thank you for your comments and
possible suggestions.
-Guy

**Comment by harrisgm** [comment]
To follow up with my earlier comments regarding to the package's use of
custom annotation files:

I have determined that the annotation package "pd.mta.1.0" contains the
"locus_type" information that is desired.
This only leaves the probe sequence information that is needed to determine
each probe's GC count (for bkg-correction).

If bioconductor had 'probe sequence' packages for the ClariomD/S style of
arrays, I would be able to run the package without the need for any outside
annotation files.

I would need packages like this example for mouse gene style arrays:
https://www.bioconductor.org/packages/release/data/annotation/html/mogene10stv1probe.html
)

I can create the probe sequence annotation data packages as described above
using "AnnotationForge" and the affymetrix "probe_tab" files.

If I create these probe sequence annotation packages, will they be easy to
upload to Bioconductor's annotation data repository?


I will be able to address all package dependencies on custom annotation
files if "probe sequence" annotation packages are added as well.

Thanks for your time.
-Guy




I can make our package work using bioconductor annotation packages

**Comment by LiNk-NY** [comment]
Hi Guy, @harrisgm 
I'm no expert in probe sequences. I would rather refer you to Jim @jmacdon. 
He would likely be able to answer your question. 

Best regards,
Marcel

**Comment by jmacdon** [comment]
The only question I see here is whether or not @harrisgm can upload the probe files to the annotation repository. This is where all the other probe packages reside, so I don't really see a problem with doing so. An alternative would be to put the probe packages on the AnnotationForge repository. @lshep @mtmorgan @dvantwisk Any druthers as to where these should go?

**Comment by LiNk-NY** [comment]
I'll let others chime in..

I have one question though: 
Would it be possible to coordinate with the maintainer of `pd.mta.1.0` and
include GC counts for probes already in the the data package? 

-MR

**Comment by jmacdon** [comment]
It could be done, but the bgp file which contains the probe sequences isn't currently parsed by pdInfoBuilder. So to do that would require someone to write a parser, generate the GC counts, and then add a table to the pdInfoPackage that is unrelated to anything that oligo does.

Which seems unlikely to happen, given the incentives. I could envision a different scenario where @harrisgm generates a data package that contains the GC content for all the probes for all the Affy arrays that Sscore2 supports. That way there is no reliance on others to do things to support Sscore2, and the data package would probably be pretty light, and once built probably never need updating.

**Comment by LiNk-NY** [comment]
Thanks for your expert input Jim @jmacdon 
It looks like the second option is best.
The question of whether to place the probe files in the annotation repository or to create a probe package with `AnnotationForge` still remains... @mtmorgan @lshep @dvantwisk

**Comment by LiNk-NY** [comment]
Hi HarrisGM @harrisgm 
I've spoken with Lori and Martin and it seems like `AnnotationHub` submission would be the ideal option. 
Please see the guidance here: https://bioconductor.org/packages/devel/bioc/vignettes/AnnotationHub/inst/doc/CreateAnAnnotationPackage.html

Feel free to ask questions on this thread. 

Best,
Marcel

**Comment by harrisgm** [comment]
Dear all,
Thank you all very much for looking into this for me.

I began by using AnnotationForge to create a "mta10probe" package from the
probe sequences in "probe_tab" file for the MTA-1.0 arrays. (
https://github.com/harrisgm/mta10probe)
However, the probe_tab file does not contain probe sequence information for
all probes on the chip.

Much to my embarrassment, I finally found out how to access the probe
sequence information contained in "pd.mta.1.0".  So this package does in
fact have everything I need to generate probeFiles for the Sscore2
application. I also have been able to easily add the "hta2.0" functionality
to the code using the "pd.hta.2.0" package.

So, I was incorrect when I said I needed annotations outside of what
BioConductor provides.  I am sorry for this mistake.

However, I do still need to generate 'probeFile' data.tables for the
Sscore2 from the bioconductor sources.  I have written a function to create
and save these probeFiles as .rda files.

Is it acceptable to save these .rda files to the /data/ directory of the
already installed R Package? The probeFiles would be generated as needed
for the given array/chip type and then retroactively placed in the R
package installation 'data' directory. This is easy enough when the package
is installed in the user library.

OR... should I use annotationHub to upload these probeFiles as annotation
packages on Bioconductor?  All information in the probeFile is sourced from
exclusively from the  "pd.mta.1.0", "pd.hta.2.0", etc.

Thanks again for everyone's help.
-Guy

**Comment by harrisgm** [comment]
Dear Bioconductor,

Regarding package submission #1190 (Sscore2):
The package has been resubmitted, due to size issues, as: submission #1274
(GCSscore)

This is Guy Harris again and I do believe I have finally corrected the
major issues with package submission #1190 (original title: "Sscore2")

Due to .git pack tracking of the previous large 'probeFiles' that I
included in "extdata", I removed the originally submitted package from
github yesterday.

I am re-submitting my package from a fresh repository, and it is now titled
"GCSscore" instead of "Sscore2".

A list of major changes:
1. all "probeFiles" for running the GCSscore algorithm are now generated
"on the fly" using only Bioconductor packages: platform design (pd) package
AND annotation .db files.
       a) Ultimately, the "probeFile" packages are created using
makeProbePackage() from 'AnnotationForge', along with a modified versions
getProbeDataAffy(), depending on the chip technology.
       b) The 'probeFile' packages that are generated for individual chip
types could be uploaded as Annotation packages as well.

2. The issues "exports" and "imports" and "depends" have been corrected.

3. It passes all R CMD check (1 NOTE for 11Mb package size) and R CMD
biocCheck (no ERRORS or WARNINGS)

4. I am adding the web hook to this new package submission my additional
edits can be viewed by you guys.


Hopefully, the resubmission of the package isn't a problem!

Thanks for all the advice, it really helped me improve this package.
-Guy
